packages/tpetra/core/src/Tpetra_BlockCrsMatrix_def.hpp

```diff
@@ -2609,6 +2609,10 @@ class GetLocalDiagCopy {
         const auto policy =
           policy_type(numExportLIDs, 1, 1)
           .set_scratch_size(0, Kokkos::PerTeam(sizeof(GO)*maxRowLength));
+        // The following parallel_for needs const access to the local values of src.
+        // (the local graph is also accessed on host, but this does not use WDVs).
+        getValuesHost();
```

Comment:
I guess the main goal of this is to call `WDV::modify_host`.

Instead of calling it implicitly through `getValuesHost`, wouldn't it be more readable to do so more explicitly?

Reply:
Basically yeah, but there are a few necessary things that these function do:
- Sync to the requested space (if data was modified in the other one)
- Mark modified in the requested space (unless read-only access)
- Check the reference counts to make sure a view in the other space isn't already alive

We want to do all of these things before turning off the tracking, so the WDV getView.... functions are the only way to do them all in one line.